### PR TITLE
Publish on release

### PR DESCRIPTION
**Pull request by hannesa2**
It's a follow up of https://github.com/eirnym/js2p-gradle/issues/30#issuecomment-786453062

### CODE REVIEW

**Comment by hannesa2** [comment]
Maybe after merge you can set a new tag to test release job ?

**Comment by eirnym** [comment]
Hey, It's a nice feature, thank you.

I need to learn how to manage gradle secrets properly

**Comment by hannesa2** [comment]
Here you set a variable in action to System environment THESECRET. `YOUR_SECRET` comes from Github settings-Secrets

        env:
          THESECRET: ${{ github.YOUR_SECRET }}

**Comment by eirnym** [comment]
could be, I need to investigate this how it's done in another gradle plugins

**Comment by eirnym** [comment]
@hannesa2 please contact me ... somehow

**Comment by eirnym** [comment]
@hannesa2 could you please contact me?

**Comment by hannesa2** [comment]
> @hannesa2 could you please contact me?

No clue how

image

In git history you see my email

**Comment by eirnym** [comment]
I found that secrets could easily be passed this way. https://github.com/ben-manes/gradle-versions-plugin/blob/master/.github/workflows/deploy.yml 

So let's upgrade our GitHub release action to this and test it

**Comment by eirnym** [comment]
@hannesa2 I've updated your PR, please review. Just after merge we'll check it

**Comment by hannesa2** [comment]
I want to avoid such commits, that's why I rebased it again

image

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -31,3 +35,28 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           VERSION: ${{ github.ref }}
+  publish:
```
Comment (hannesa2) [inline .github/workflows/release.yml:38]: If you do it like this, this makes `build` almost obsolete and increase overall runtime.
I will make a proposal to solve this. If you don't like it, simply revert it

Comment (eirnym) [reply .github/workflows/release.yml]: I agree with you and probably we should put publishing call just after tagging, shouldn't we?

I followed logic, that version is given from a tag, and we put a tag _after_ build. So if we do a publish before tagging, an old version will be published.

**Comment by eirnym** [comment]
I like a new version, let's try it

File: .github/workflows/release.yml (hunk 1 above)
Comment (hannesa2) [reply .github/workflows/release.yml]: When  I understand right:
No, you do a `git tag -a 1.x.s`omething` -m "something" && git push --tags` this creates the tag, and **after** this the CI starts to run

**Comment by hannesa2** [comment]
Normally, I do a test run on personal fork, but we can test it as well. With admin right, we can remove release 😉

**Pull request merged**